dubbo-common/src/main/java/org/apache/dubbo/config/AbstractInterfaceConfig.java

```diff
@@ -221,7 +221,7 @@ public AbstractInterfaceConfig(ModuleModel moduleModel) {
     /**
      * The url of the reference service
      */
-    protected final transient List<URL> urls = new ArrayList<URL>();
+    protected final transient List<URL> urls = new ArrayList<>();
```

Comment:
**Suggestion**: Ensure that the simplification of conditionals and the refactoring performed do not alter the intended logic, especially in critical configuration handling. <br> **Code Suggestion**: 
 ```
 - protected final transient List<URL> urls = new ArrayList<URL>();
 + protected final transient List<URL> urls = new ArrayList<>();
 - if (registryConfig.getScopeModel() != applicationModel) {
 + if (registryConfig != null && registryConfig.getScopeModel() != applicationModel) {
 - if (methodConfigs != null && methodConfigs.size() > 0) {
 + if (methodConfigs != null && !methodConfigs.isEmpty()) {
 + if (StringUtils.isEmpty(getGroup()) && interfaceConfig != null) {
 +     return interfaceConfig.getGroup();
 + }
 + return getGroup();
 + if (StringUtils.isEmpty(getVersion()) && interfaceConfig != null) {
 +     return interfaceConfig.getVersion();
 + }
 + return getVersion();
 ```